src/types/params.ts

```diff
@@ -80,8 +80,9 @@ export type ExtractPathParamType<
 export type ExtractRouteParamTypes<TRoute> = TRoute extends {
   path: { params: infer PathParams extends Record<string, Param> },
   query: { params: infer QueryParams extends Record<string, Param> },
+  host: { params: infer HostParams extends Record<string, Param> },
 }
-  ? ExtractParamTypes<MergeParams<PathParams, QueryParams>>
+  ? ExtractParamTypes<MergeParams<HostParams, MergeParams<PathParams, QueryParams>>>
```

Comment:
This made me wonder if we even need this `MergeParams` utility anymore. Can't this just be `HostParams & PathParams & QueryParams`?

Reply:
that does seem to be true. Why did we need it before? 🤔

Reply:
I think that's just a relic of previous types.